File: pset6/readability/readability.py
from cs50 import get_string


# User's text input
text = get_string("Text: ")

i = 0
j = 1
k = 0
l = 0

for i in range(len(text)):
    if (ord(text[i]) > 64 and ord(text[i]) < 91) or (ord(text[i]) > 96 and ord(text[i]) < 123):
        # Count the letters
        k = k + 1

    # Count the words
    if text[i] == ' ':
        j = j + 1

    if text[i] in ['!', '.', '?']:
        # Count the sentences
        l = l + 1

# Coleman-Liau index
index = 0.0588 * k / j * 100 - 0.296 * l / j * 100 - 15.8

if (index < 1):
    # Grades smaller than 1
    print("Before Grade 1")
elif (index > 15):
    # Grades bigger than 16
    print("Grade 16+")
else:
    # The rest
    print(f"Grade {int(round(index, 0))}")

File: pset2/readability/readability.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{

    string text = get_string("Text: ");             // User's text input

    char c;
    int i = 0;
    int j = 0;
    int k = 0;
    int l = 0;

    float index;

    do
    {
        c = text[i];
        i++;

        if ((c > 64 && c < 91) || (c > 96 && c < 123))  // Count the letters
        {
            k++;
        }

        if (c == ' ' || c == '\0')                      // Count the words
        {
            j++;
        }

        if (c == '!' || c == '.' || c == '?')           // Count the sentences
        {
            l++;
        }
    }
    while (c != '\0');

    index = 0.0588 * (float) k / j * 100 - 0.296 * (float) l / j * 100 - 15.8;      // Coleman-Liau index

    if (index < 1)
    {
        printf("Before Grade 1\n");         // Grades smaller than 1
    }
    else if (index > 15)                    // Grades bigger than 16
    {
        printf("Grade 16+\n");              // The rest
    }
    else
    {
        printf("Grade %0.f\n", index);
    }

}